From a0cb4b0989127c1ee820c9a7c1db35253c0486a1 Mon Sep 17 00:00:00 2001
From: Souma-Sumire <553469159@qq.com>
Date: Sat, 23 Aug 2025 23:47:51 +0800
Subject: [PATCH] PostNamazu

---
 cactbot/types/event.d.ts | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/cactbot/types/event.d.ts b/cactbot/types/event.d.ts
index d99c42d18..34dc54879 100644
--- a/cactbot/types/event.d.ts
+++ b/cactbot/types/event.d.ts
@@ -524,6 +524,30 @@ type CactbotChooseDirectoryHandler = (msg: {
   call: 'cactbotChooseDirectory';
 }) => { data: string } | undefined;
 
+export type PostNamazuCall =
+  | 'DoTextCommand'
+  | 'command'
+  | 'DoWaymarks'
+  | 'place'
+  | 'mark'
+  | 'DoInsertPreset'
+  | 'preset'
+  | 'DoQueueActions';
+
+type PostNamazuHandler = (msg: {
+  call: 'PostNamazu';
+  c: PostNamazuCall;
+  p: string;
+  d?: number;
+}) => Promise<void>;
+
+type GetLanguageHandler = (msg: { call: 'getLanguage' }) => {
+  language: string;
+  languageId: string;
+  region: string;
+  regionId: string;
+};
+
 export type OverlayHandlerAll = {
   'broadcast': BroadcastHandler;
   'subscribe': SubscribeHandler;
@@ -537,6 +561,8 @@ export type OverlayHandlerAll = {
   'cactbotSaveData': CactbotSaveDataHandler;
   'cactbotLoadData': CactbotLoadDataHandler;
   'cactbotChooseDirectory': CactbotChooseDirectoryHandler;
+  'postNamazu': PostNamazuHandler;
+  'getLanguage': GetLanguageHandler;
 };
 
 export type OverlayHandlerTypes = keyof OverlayHandlerAll;
-- 
2.45.1.windows.1

