From 94e8f3b926c5ec08b79af18275325e46453df76d Mon Sep 17 00:00:00 2001
From: Souma-Sumire <553469159@qq.com>
Date: Sun, 26 Oct 2025 19:33:17 +0800
Subject: [PATCH] party type

---
 types/event.d.ts | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/cactbot/types/event.d.ts b/cactbot/types/event.d.ts
index d99c42d18..808c2d645 100644
--- a/cactbot/types/event.d.ts
+++ b/cactbot/types/event.d.ts
@@ -7,12 +7,29 @@ declare global {
     'onOverlayStateUpdate': CustomEvent<{ isLocked: boolean }>;
   }
 }
+export enum PartyType {
+  Solo,
+  Party,
+  AllianceA,
+  AllianceB,
+  AllianceC,
+  AllianceD,
+  AllianceE,
+  AllianceF,
+}
+
 export interface Party {
   id: string;
   name: string;
   worldId: number;
   job: number;
   inParty: boolean;
+  contentId: number;
+  flags: number;
+  level: number;
+  objectId: number;
+  partyType: PartyType;
+  territoryType: number;
 }
 
 export type BardSongType = 'Ballad' | 'Paeon' | 'Minuet' | 'None';
-- 
2.45.1.windows.1

